library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -82,13 +84,41 @@ def self.configured?
       @configured = out["exit"] == 0
     end
 
-    # Creates a new 'single' snapshot
+    # Returns whether creating the given snapshot type is allowed
+    # Information is taken from Linuxrc (DISABLE_SNAPSHOTS)
+    #   * "all" - all snapshot types are temporarily disabled
+    #   * "around" - before and after calling YaST
+    #   * "single" - single snapshot at a given point
+    #
+    # @param [Symbol] one of :around (for :post and :pre snapshots) or :single
+    # @return [Boolean] if snapshot should be created
+    def self.create_snapshot?(snapshot_type)
+      disable_snapshots = Yast::Linuxrc.get_value(Yast::LinuxrcClass::DISABLE_SNAPSHOTS)
+
+      # Feature is not defined on Linuxrc commandline
+      return true if disable_snapshots.nil? || disable_snapshots.empty?
+
+      disable_snapshots = disable_snapshots.downcase.tr("-_.", "").split(",").map(&:to_sym)
+
+      if [:around, :single].include?(snapshot_type)
+        return false if disable_snapshots.include?(:all)
+        return !disable_snapshots.include?(snapshot_type)
+      else
+        raise ArgumentError, "Unsupported snapshot type #{snapshot_type.inspect}, " \
+          << "supported are :around and :single"
+      end
+    end
+
+    # Creates a new 'single' snapshot unless disabled by user
     #
     # @param description [String] Snapshot's description.
     # @return [FsSnapshot] The created snapshot.
     #
     # @see FsSnapshot.create
+    # @see FsSnapshot.create_snapshot?
     def self.create_single(description)
+      return nil unless create_snapshot?(:single)
```

Comment:
No, I could not put it into `create` method, because self.create_post searches for `previous` snapshot first. So, to keep consistency with the other methods, it's three times "almost" the same code.